src/libpmemfile/preload.c

```diff
@@ -60,6 +60,29 @@
 #include "libpmemfile-core.h"
 #include "util.h"
 
+/*
+ * Some syscalls that are missing on older kernels.
+ */
+#ifndef SYS_renameat2
+#define SYS_renameat2 316
+#endif
+
+#ifndef SYS_execveat
+#define execveat 322
+#endif
+
+#ifndef SYS_copy_file_range
+#define SYS_copy_file_range 326
+#endif
+
+#ifndef SYS_preadv2
+#define SYS_preadv2 327
+#endif
+
+#ifndef SYS_pwritev2
+#define SYS_pwritev2 328
+#endif
```

Comment:
I don't see anything in libpmemfile that's x86-64 specific. It's x86-64 only because libsyscall_intercept is.
At some point in the future someone might want to port it to other architecture and I think giving them nice compile error instead of silent runtime breakage is much better approach...